api/src/controllers/namespace.ts

```diff
@@ -0,0 +1,131 @@
+//
+// Copyright © 2020 Province of British Columbia
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// Created by Jason Leach on 2020-04-27.
+//
+
+'use strict';
+
+import { errorWithCode, logger } from '@bcgov/common-nodejs-utils';
+import { Request, Response } from 'express';
+import { difference } from 'lodash';
+import config from '../config';
+import DataManager from '../db';
+
+export const createNamespace = async (req: Request, res: Response): Promise<void> => {
+  const dm = new DataManager(config);
+  const { NamespaceModel } = dm;
+  const { profileId } = req.params;
+  const { body } = req;
+  const aBody = { ...body, profileId };
+
+  const diff = difference(NamespaceModel.requiredFields, Object.keys(aBody));
+  if (diff.length !== 0) {
+    throw errorWithCode(`Missing required properties: ${diff}`, 400);
```

Comment:
does that output correctly or `properties:  {Object object}`?  if not perhaps `diff.join()`

Reply:
I've tested it and it seem to be working as expected.